deps/rabbit/src/rabbit_fifo.erl

```diff
@@ -1500,14 +1521,30 @@ return_all(Meta, #?MODULE{consumers = Cons} = State0, Effects0, ConsumerId,
                 end, {State, Effects0}, Checked).
 
 %% checkout new messages to consumers
-checkout(#{index := Index} = Meta, OldState, State0, Effects0) ->
+checkout(Meta, OldState, State, Effects) ->
+    checkout(Meta, OldState, State, Effects, true).
+
+checkout(#{index := Index} = Meta, #?MODULE{cfg = #cfg{resource = QName}} = OldState, State0,
+         Effects0, HandleConsumerChanges) ->
     {State1, _Result, Effects1} = checkout0(Meta, checkout_one(Meta, State0),
                                             Effects0, {#{}, #{}}),
     case evaluate_limit(Index, false, OldState, State1, Effects1) of
         {State, true, Effects} ->
-            update_smallest_raft_index(Index, State, Effects);
+            case have_active_consumers_changed(State, HandleConsumerChanges) of
```

Comment:
@dcorbacho @kjnilsson didn't we want to get rid of this on the hot delivery path?

Reply:
I thought I did, unless it got lost on the cherry-pick. Weird, I'll review it again

Reply:
Sorry, that's right. Now we only apply it if the `HandleConsumerChanges` flag is set to `true`, excluding all enqueue/return/credit/settle/discard operations - the hot delivery path. It's only on checkout/node up/node down/config updates that we notify the decorators